### ISSUE TITLE
Collection-set in mastering step still adding step name as collection

### ISSUE THREAD

**Issue by aajacobs** [marklogic/marklogic-data-hub#2482]
### Welcome Hub enthusiast!

Let's get the most out of this ticket.

#### The issue

Short description of the problem:  in DHF5, when using a mastering step with a set-collections merge option, the mastering step name is also added as a collection

What behavior are you expecting?  To only have the collection specified in the merge config.

#### Tech details

Which Operating System are you using?  CentOS

Which version of MarkLogic are you using?  9.0-9.1

Which version of the Data Hub Framework are you using? 5.0rc1

Is this a QuickStart UI bug? If so, which browser are you using?  No. But chrome.

So given the following flow (note the mastering step name):
![image](https://user-images.githubusercontent.com/8723378/57527764-83ccaf80-72fe-11e9-9686-8b648fff0cf1.png)

And the following merge options:
![image](https://user-images.githubusercontent.com/8723378/57527771-87603680-72fe-11e9-8a06-532e98cdf60e.png)

My merged docs are in both the "MasterPerson" collection (name of the step) as well as the collection I defined--"MasterPersonRecords".  

![image](https://user-images.githubusercontent.com/8723378/57527928-e7ef7380-72fe-11e9-896c-af6a938c323f.png)

But it's the same for any doc processed by the step--it has the collection I specified PLUS the step name collection.  I would expect with the "collection-set" merge option for only those collections to be added.

**Comment by aebadirad** [comment]
@aajacobs  can you verify this bug still exists after a formal upgrade to official 5.0.0?

**Comment by aajacobs** [comment]
@aebadirad Yes, this behavior still exists in 5.0.0.  In fact, it also adds the entity name as a collection as well where rc1 did not.  So to recap, collection set adds as collections (1) the entity name, (2) the mastering step name, and (3) the value of the collection set in the merge options.

**Comment by aajacobs** [comment]
sample project I used is available at https://github.com/aajacobs/dhf5-sm-collections

**Comment by rjrudin** [comment]
Did some testing in the dh-5-example project, which is still using the non-HC format for merge options. 

When I added mergeSet1 and mergeSet2 as "set" collections for onMerge, the merged document ends up with - mastered1, mergeSet1, mergeSet2, and http://marklogic.com/Order-0.0.1/Order . So the step name is not being added. 

Did a test with "add", and it worked correctly - again not including the step name.

So not sure when this was fixed, but it appears that the step name is not being applied, at least in DHF 5.4. Was likely fixed much earlier.
